sycl/source/detail/queue_impl.hpp

```diff
@@ -59,11 +59,7 @@ class queue_impl {
              const property_list &PropList)
       : queue_impl(Device,
                    detail::getSyclObjImpl(
-                       context(createSyclObjFromImpl<device>(Device), {},
```

Comment:
It seems the patch removes support for property::context::cuda.  Could you please restore it? 

Probably this property should be changed to be a property of context. So far, we probably could create a new context if use_primary_context is passed.

Reply:
So I can't find where this "if" could possibly be true.

As far as I can tell, the only two mentions of DefaultContextType are in this file, and it's constexpr assigned to a different value than the one checked in the if.